Remove B code (ISO 639-2)

KuwaLee opened the pull request:
After seening https://github.com/bluesky-social/social-app/pull/4313, I found that language code contains both B and T codes (`ISO 639-2`), which means there are many duplicated langusge code, for example: 

`{code3: 'fra', code2: 'fr', name: 'French'}` 
`{code3: 'fre', code2: 'fr', name: 'French'}`

 It should use one standard, B or T, not both. (And UI fail to show both, I think it might because  `code2` property is duplicated )

According to [wikipedia](https://en.wikipedia.org/wiki/ISO_639-2#B_and_T_codes) T code are more favored, thus this PR remove B language code with [this reference](https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes).
![image](https://github.com/bluesky-social/social-app/assets/30096511/bb24188f-5ead-41e8-b93f-a891cec55116)

Plus, I would like to know if its support to use region in `code2` for example `zh-CN` and `zh-TW,` or it stick strictly only `ISO 639-2`?

KuwaLee committed:
Remove B code (ISO 639-2)

KuwaLee commented:
And I find the ultimate answer why feed with lang filter malfunction, after removing duplicated language code, it work with chinese (which `chi/zho` are duplicated for `zh`)

KuwaLee commented:
This did fix to some degree in lang filter malfunction (For example Chinese which chi/zho is duplicated for zh), tested by me and @auroursa 

Before:
![image](https://github.com/bluesky-social/social-app/assets/30096511/c1ea9fd1-9807-4bd4-99ac-11104fdadef6)
After:
![image](https://github.com/bluesky-social/social-app/assets/30096511/47a4151f-b6b1-4a7a-8111-29c4c5ad94a8)

surfdude29 commented:
I think the goal of this PR is good but I think the current order of languages that the user sees in the interface should be retained.

For example, a user looking for Dutch; Flemish would expect to find it under D not N (because even though the list is sorted by `code3`, those correspond in large part to the language name in English).

So I suggest instead that the `code3` in the B code entry should be replaced with the equivalent T code, with the current list order maintained.

surfdude29 commented:
To illustrate what I mean, instead of being deleted line 65 would become:
`{code3: 'sqi', code2: 'sq', name: 'Albanian'},`
and line 504/486 would be deleted instead

KuwaLee commented:
@surfdude29 by default it seems it order by `code3`, and I think it might need to discuss with bluesky team to reorder by `name` or reorder during runtime.

surfdude29 commented:
> @surfdude29 by default it seems it sorted by `code3`, and I think it might need to discuss with bluesky team to reorder by `name` or reorder during runtime.

@KuwaLee Ah OK, I think that change should probably be merged before this one then as otherwise it could cause quite a bit of confusion if languages seem to disappear from the list.

Also, one other potential issue that occurred to me: for users that have one of these deleted `code3` languages set as their Primary Language or one of their Content Languages, how will this PR affect them? Will their language settings be automatically migrated?

KuwaLee commented:
@surfdude29 AFAIK bluesky seems are using code2 only now

surfdude29 commented:
> (And UI fail to show both, I think it might because code2 property is duplicated )

Both are shown in the Primary Language dropdown for me (but not in the Content Languages dialog), so this PR would fix that issue too.

KuwaLee commented:
@surfdude29 Primary Language are only use when translating post

surfdude29 commented:
> @surfdude29 AFAIK bluesky seems are using code2 only now

@KuwaLee code3 can be used for determining what languages are shown in feeds and for the Primary Language setting, right? Or else, why are they available as options? 🤔

KuwaLee commented:
@surfdude29 no, `code2` are actually added to post , and what used to filter post in feed also `code2`

surfdude29 commented:
> @surfdude29 Primary Language are only use when translating post

@KuwaLee Yes, but I was meaning that this PR would fix the issue of languages being displayed twice in the UI in the Primary Language dropdown :)

For example:
![IMG_6073](https://github.com/bluesky-social/social-app/assets/149612116/5fdbaa26-0f13-4b64-bae8-703f897cda8c)

KuwaLee commented:
@surfdude29 that is simply because it have 2 object, Primary Language is used for parameters when calling google translation

And  what lang of post is what you choose when you posting, And they really added only code2
![image](https://github.com/bluesky-social/social-app/assets/30096511/97a838a8-72c9-4444-aa43-beea8b17980f)
![image](https://github.com/bluesky-social/social-app/assets/30096511/3051abc2-ffe5-418a-8a98-2b19bf7dedfe)
![image](https://github.com/bluesky-social/social-app/assets/30096511/43ae25d4-25bb-4c8e-97fa-acbce134cbf5)

surfdude29 commented:
@KuwaLee I apologise for my confusion, it seems I totally misunderstood how languages work on Bluesky, oops 😬

KuwaLee committed:
Regen code for LANGUAGES and sorted by name.

Regen code from www.loc.gov, and sorted by name, excluding 'qaa-qtz', which is "Reserved for local use".
